packages/react/src/slider/root/SliderRoot.tsx

```diff
@@ -157,28 +158,25 @@ export namespace SliderRoot {
     values: ReadonlyArray<number>;
   }
 
-  export interface Props
+  export interface Props<TValue extends number | number[]>
```

Comment:
My first approach here was to try to carry this generic all the way down in the slider components but it resulted to just pollute  the code with generics.
Instead I simple just try to modify the top level component that the consumer will use.

Reply:
@seloner 
I believe we can define Props without using generics. By utilizing intersection types, we can potentially make the necessary changes internally without affecting the external API. This approach could simplify the type definitions and improve readability.

Here’s an example of how we can implement this:

```diff
@@ -158,7 +158,10 @@ export namespace SliderRoot {
     values: ReadonlyArray<number>;
   }
 
-  export interface Props<TValue extends number | number[]>
+  export type Props = PropsTemplate<number> &
+    PropsTemplate<number[]> &
+    PropsTemplate<number | number[]>;
+  export interface PropsTemplate<TValue extends number | number[]>
     extends Pick<
         useSliderRoot.Parameters,
         | 'disabled'
```

Please refer to https://github.com/mui/material-ui/pull/44777#issuecomment-2564268790